lib/jxl/encode.cc

```diff
@@ -220,6 +222,15 @@ JxlEncoderStatus JxlEncoderSetBasicInfo(JxlEncoder* enc,
     default:
       return JXL_ENC_ERROR;
   }
+  // Adjusting the number of extra channels; SetAlphaBits sets this to 1
+  // if alpha is present, so info->num_extra_channels should be provided
+  // as at least one in that case.
+  // if (enc->metadata.m.num_extra_channels > info->num_extra_channels){
+  //  return JXL_API_ERROR("Invalid value for number of extra channel");
+  //}
+  enc->metadata.m.num_extra_channels += info->num_extra_channels;
```

Comment:
I would consider alpha to be an extra channel like any other, even if it has the special property that there are special functions for it and it can be passed to the encoder / received from the decoder together with the color image using RGBA buffers. In the spec (and in the C++ code), an alpha channel is treated like any other extra channel. So let's try to avoid doing +1 or -1 on num_extra_channels in the API.

Reply:
To me it looks ok to require setting num_extra_channels in the info to 1. Note that this is not backwards compatible, in any application using the encoder where info.alpha_bits is used, num_extra_channels will have to be set correctly as well.

It would be nice, though, to avoid having to call JxlEncoderSetExtraChannelInfo for a default alpha channel. You should be able to call it if you want to, but not be required to: if you set alpha_bits in the basic info, it's implied that there is an alpha channnel.

We decided that for simplicity, this has to be the extra channel with index 0.

WDYT?

Reply:
Sounds good to me to have a special case for a default alpha channel, configured from the basicinfo.

I would prefer to be strict on `num_extra_channels`, so we have the invariant that the total number of channels is always equal to `num_color_channels + num_extra_channels`. This is something that you'll likely often want to know, and if we're not strict on this, then the shortest expression that gives you the total number of channels would be something like `num_color_channels + num_extra_channels + (num_extra_channels == 0 && alpha_bits > 0 ? 1 : 0)` which feels clumsy.

Reply:
Make sense, now I have changed to code such that it is required to always set `num_extra_channels` correctly.